experiment/popularity_community/crawl_torrents.py

```diff
@@ -0,0 +1,214 @@
+"""This script crawl first 100 torrens from random nodes in the network.
+
+### Usage
+
+```
+export PYTHONPATH=${PYTHONPATH}:`echo ../../../src/{pyipv8,tribler-common,tribler-core} | tr " " :`
+
+python3 crawl_torrents.py [-t <timeout_in_sec>] [-f <output_file.json>]
+```
+
+Where:
+* `timeout_in_sec` means a time that the experiment will last
+* `output_file.json` means a path and a result file name
+
+### Example
+
+```
+python3 crawl_torrents.py -t 600
+python3 crawl_torrents.py -t 600 -f result.json
+```
+
+"""
+import argparse
+import asyncio
+import json
+import logging
+import os
+from collections import defaultdict
+
+import lz4
+
+from ipv8.lazy_community import lazy_wrapper
+from ipv8.peer import Peer
+from ipv8.peerdiscovery.discovery import RandomWalk
+from tool.tiny_tribler_service import TinyTriblerService
+from tribler_core.modules.metadata_store.community.remote_query_community import RemoteQueryCommunity, \
+    SelectResponsePayload, SelectRequest, RemoteSelectPayload
+from tribler_core.modules.metadata_store.serialization import REGULAR_TORRENT
+from tribler_core.utilities.unicode import hexlify
+
+UNLIMITED = -1  # this magic number is use inside many Tribler modules
+
+IPV8_WALK_INTERVAL = 0.05
+
+RANDOM_WALK_TIMEOUT_IN_SEC = 5
+RANDOM_WALK_WINDOW_SIZE = 0
+RANDOM_WALK_RESET_CHANCE = 30
+
+CRAWLER_REQUEST_TIMEOUT_IN_SEC = 60
+CRAWLER_SAVE_INTERVAL_IN_SEC = 60
+
+
+class CrawlerSelectRequest(SelectRequest):
+    """Custom class for RemoteQueryCommunity select request
+
+    It has been added only for having an ability to increase `timeout_delay`
+    """
+
+    @property
+    def timeout_delay(self):
+        return CRAWLER_REQUEST_TIMEOUT_IN_SEC
+
+
+class Crawler(RemoteQueryCommunity):
+    """ Crawler based on RemoteQueryCommunity
+
+    The approach based on two ideas:
+        * use "aggressive" values for network walking (see ExperimentConfiguration.RANDOM_WALK_*)
+        * use RemoteQueryCommunity's ability to select MetadataInfo from remote db
+    """
+
+    def __init__(self, my_peer, endpoint, network, metadata_store, out_file_path):
+        super().__init__(my_peer, endpoint, network, metadata_store)
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self._output_file_path = out_file_path
+        self.max_peers = UNLIMITED
+        self.queried_peers_limit = UNLIMITED
+
+        self.torrent_list = defaultdict(list)
```

Comment:
The name is a bit misleading, as it is a dictionary of lists and not a list

Reply:
Right. Will fix it.

Reply:
Fixed